### PR TITLE
Skeletons can treat their bone wounds again, improves door crush dislocation fix

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
When I did #52951 and checked for pierce immunity to disallow treatments through thick clothing, I forgot about the existence of skeletons who would always fail that check. This specific can_inject() check now ignores species traits so they can still apply bone_gel and slings as needed

In addition, the "get crushed in an airlock/firelock" improvised fix for dislocated limbs is now done by a signal instead of a direct proc like it should have a while ago.

## Why It's Good For The Game
Lets bone species fix their bone wounds without surgery/milk again, improves a sinful bit of my snowflake code in door crushes

## Changelog
:cl: Ryll/Shaps
fix: Skeletons and other boneheads can now treat their bone wounds with bone gel and slings without getting messages about their flesh being impenetrable.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
image

**Comment by Ryll-Ryll** [comment]
Yeah thanks for pointing it out @ath1909, when I saw your message I spent a solid 3 minutes going back and forth between thinking I was good and that I wasn't good. Turned out I wasn't!

File: code/datums/wounds/_wounds.dm [hunk 1]
```diff
@@ -250,7 +250,10 @@
 		to_chat(user, "<span class='warning'>You're already interacting with [victim]!</span>")
 		return TRUE
 
-	if(!victim.can_inject(user, TRUE))
+	// next we check if the bodypart in actually accessible (not under thick clothing). We skip the species trait check since skellies
+	// & such may need to use bone gel but may be wearing a space suit for..... whatever reason a skeleton would wear a space suit for
+	var/mob/living/carbon/human/victim_human = victim
+	if(istype(victim_human) && !victim_human.can_inject(user, TRUE, ignore_species = TRUE))
 		return TRUE
```
Comment (Rohesie) [inline code/datums/wounds/_wounds.dm:257]:
```suggestion
	if(ishuman(victim))
		var/mob/living/carbon/human/victim_human = victim
		if(!victim_human.can_inject(user, TRUE, ignore_species = TRUE))
			return TRUE
```

File: code/datums/wounds/bones.dm [hunk 2]
```diff
@@ -199,7 +199,17 @@
...
 
-/datum/wound/blunt/moderate/crush()
+/datum/wound/blunt/moderate/wound_injury(datum/wound/old_wound)
+	. = ..()
+	RegisterSignal(victim, COMSIG_LIVING_DOORCRUSH, .proc/door_crush)
+
+/datum/wound/blunt/moderate/Destroy()
+	if(victim)
+		UnregisterSignal(victim, COMSIG_LIVING_DOORCRUSH)
+	return ..()
```
Comment (Rohesie) [inline code/datums/wounds/bones.dm:209]:
```suggestion

/datum/wound/blunt/moderate/Destroy()
	if(victim)
		UnregisterSignal(victim, COMSIG_LIVING_DOORCRUSH)
	return ..()


/datum/wound/blunt/moderate/wound_injury(datum/wound/old_wound)
	. = ..()
	RegisterSignal(victim, COMSIG_LIVING_DOORCRUSH, .proc/door_crush)

```
Try to leave the `New()`, `Initialize()` and `Destroy()` and related definitions at the top, as they set up the object's basic definition.

File: code/game/machinery/doors/door.dm [hunk 3]
```diff
@@ -349,11 +349,7 @@
 /obj/machinery/door/proc/crush()
 	for(var/mob/living/L in get_turf(src))
 		L.visible_message("<span class='warning'>[src] closes on [L], crushing [L.p_them()]!</span>", "<span class='userdanger'>[src] closes on you and crushes you!</span>")
-		if(iscarbon(L))
-			var/mob/living/carbon/C = L
-			for(var/i in C.all_wounds) // should probably replace with signal
-				var/datum/wound/W = i
-				W.crush(DOOR_CRUSH_DAMAGE)
+		SEND_SIGNAL(L, COMSIG_LIVING_DOORCRUSH)
```
Comment (Rohesie) [inline code/game/machinery/doors/door.dm:352]:
```suggestion
		SEND_SIGNAL(L, COMSIG_LIVING_CRUSHED_BY_DOOR, src)
```
* Be clear in the signal name who's the agent and who's the target of the action.
* Pass relevant information, such as the source (and the proc input args whenever they exist).

**Review by Rohesie — APPROVED** [review]
Nice code improvement

**Pull request merged**